src/assembler.cpp

```diff
@@ -1242,6 +1242,9 @@ int main(int argc, char* argv[])
                         constant.b = (uint32_t)values[0] != 0;
                         break;
                     }
+
+                    default:
```

Comment:
Ok so this part is building a register that holds a constant value, which can only be float, int or bool.
So this default case is indeed an error that should be properly reported, I added a throw statement, which seems to be the prefered way to report that kind of error in this file, but let me know if you want to me to report it another way.